fix install script

timonegk opened the pull request:
## Proposed changes
Fixes install script. Not tested yet.

## Related issues
Closes #73 and closes #81.

## Necessary checks
- [ ] Write documentation
- [ ] Create issues for future work
- [x] Test on your machine
- [ ] Test on the robot
- [x] Put the PR on our Project board

timonegk commented:
I tested the script in a fresh docker container and it works.

ftsell commented on scripts/install.pl:
```diff
@@ -94,13 +99,18 @@ ()
         or system "sudo apt update"
-        or system "sudo apt install -y python-catkin-tools python-rosdep"
+        or system "sudo apt install -y python3-pip python3-rosdep"
+    or system "sudo pip3 install git+https://github.com/catkin/catkin_tools.git"
```
Why as root? Is `pip3 install --user` not more appropriate?

timonegk replied:
Maybe. But many (especially new) users don't have `~/.local/bin` in their `PATH`, so I figured it would be easier to just install it globally (and it doesn't interfere with system packages because `pip` installs to `/usr/local/`).

ftsell replied:
hmm, alright

ftsell commented on scripts/install.pl:
```diff
@@ -134,36 +143,50 @@ ()
     chdir $start_dir;
 
-    print "Do you want to clone bitbots gogs repos ('ansible_robots', 'doku', 'basler_drivers') now?$/";
+    print "Do you want to clone the private repos ('ansible_robots', 'doku' and 'basler_drivers') now?$/";
```
It is now only called `ansible`

ftsell commented on scripts/install.pl:
```diff
@@ -134,36 +143,50 @@ ()
     if (read_yes_no_input()) {
-        system 'git clone gogs@gogs.mafiasi.de:Bit-Bots/ansible_robots.git'
+    if (! -e "ansible") {
+        system 'git clone git@git.mafiasi.de:Bit-Bots/ansible.git'
```
I personally have ansible cloned into bitbots_tools because there it is registered in the .gitigignore file. We might want to do the same here too

ftsell commented on scripts/install.pl:
```diff
@@ -134,36 +143,50 @@ ()
-        system 'git clone gogs@gogs.mafiasi.de:Bit-Bots/doku.git'
-            and print "Could not clone doku$/$/";
+    if (! -e "doku") {
+            system 'git clone git@git.mafiasi.de:Bit-Bots/doku.git'
```
I personally have the internal docs cloned into bitbots_tools as bitbots_docs_internal because it is registered in the .gitignore file there. We might want to do the same here too

timonegk commented:
I added a cleanup of the `scripts` folder and the `Makefile`. I removed the `make build` because I never heard that anybody used it and removed the scripts `build.sh` (for the same reason), `create_venv.bash` (also unused), `install_py_extension.bash` (is now done via cmake) and `repair.sh` (fortunately not needed anymore).
